### DISCUSSION TITLE
Discontinuities in DG solution only at specific interfaces.

### DISCUSSION THREAD

**Discussion by hadato** [mfem/mfem#4628] [Q&A]
Hello everyone. I have been implementing a simple heat solver and would like it to be able to handle interfacial jumps in the solution. I have previously implemented the CG method, however, per definition, the solution is continuous.
Recently, I have been looking to the DG method in a hope I could solve my problem. 

The simplest physical example would be a rod composed of two different materials featuring and an interface with an interfacial resistance.  If one side is kept at higher temperature, there will be heat flux through the wire (and the interface), resulting in a temperature jump at the interface. The solution in the domains is continuous. 

The problem is as follows: 

$$\frac{\partial T}{\partial t} - \kappa \Delta T = 0$$

or in the weak form

$$\int_{\Omega}\frac{\partial T}{\partial t} w dx + \int_{\Omega} \kappa \nabla T \cdot \nabla w  dx - \int_{\Gamma} \widehat{F}  w  d\Gamma=0,$$

which leas to a simple use of the MassIntegrator for the first term and variations of the DiffusionIntegrator throughout the solver for the Dirichlet BC and the solution continuity (from examples 14, 16). 

I could simply weakly enforce the continuity by adding the  AddInteriorFaceIntegrator(new DGDiffusionIntegrator($\kappa$, 0, $\beta$)), where $\beta$ is a penalty parameter. This represents the term:

$$\hat{F} = \{-\kappa \nabla T \cdot \mathbf{n}\} - \beta[T]$$

The problem is that this term is not valid at the interfaces as I need to introduce the jumps. What I read, it should be enough to change $\beta$ in the previous expression to account for the interfacial resistance $R$. 

$$\hat{F} = \{-\kappa \nabla T \cdot \mathbf{n}\} - \frac{1}{R}[T]$$

[T] is the temperature difference at the interface (or the neighboring elements at the same node)  $[T] = T^+ - T^-$. 

Is there a method how to include the discontinuity at specific internal interfaces in the DG method?

Has someone encountered a similar problem before? Eventually, is there a way how to introduce jumps in the solution using the CG method?

Any help would be highly appreciated! 
Tomas

**Comment by tzanio** [comment]
Hi Tomas,

Do I understand correctly that you want a discretization space for $T$ that is generally continuous, except on specific faces, where it is discontinuous? Such partially discontinuous spaces have been of interest to us, but we don't currently support them in the code. 

While not optimal, maybe it is possible to consider the fully discontinuous (L2/DG) space in your code and impose continuity on most faces either strongly (through an interpolation/constraint matrix similar to AMR/Hybridization) or weakly (by adding the jumps to the bilinear form and penalizing them with a large number).

That being said, we do want to add partially discontinuous spaces to the code for other reasons, which will be a much better solution.

Hope this helps,
Tzanio
